config/eventhub/config.go

```diff
@@ -0,0 +1,103 @@
+	p.AddResourceConfigurator("azurerm_eventhub_namespace", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.Kind = "EventHubNamespace"
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Eventhub/namespaces/namespace1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Eventhub",
+			"namespaces", "name",
+		)
+		r.LateInitializer = config.LateInitializer{
+			IgnoredFields: []string{"network_rulesets"},
+		}
+	})
+
+	p.AddResourceConfigurator("azurerm_eventhub", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.Kind = "EventHub"
+		r.References = config.References{
+			"namespace_name": config.Reference{
+				Type: "EventHubNamespace",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.EventHub/namespaces/namespace1/eventhubs/eventhub1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Eventhub",
+			"namespaces", "name",
+			"eventhubs", "name",
+		)
+	})
+
+	p.AddResourceConfigurator("azurerm_eventhub_consumer_group", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.Kind = "ConsumerGroup"
+		r.References = config.References{
+			"namespace_name": config.Reference{
+				Type: "EventHubNamespace",
+			},
+			"eventhub_name": config.Reference{
+				Type: "EventHub",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.EventHub/namespaces/namespace1/eventhubs/eventhub1/consumerGroups/consumerGroup1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Eventhub",
+			"namespaces", "name",
+			"eventhubs", "name",
+			"consumergroups", "name",
+		)
+	})
+
+	p.AddResourceConfigurator("azurerm_eventhub_authorization_rule", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.Kind = "AuthorizationRule"
+		r.References = config.References{
+			"namespace_name": config.Reference{
+				Type: "EventHubNamespace",
+			},
+			"eventhub_name": config.Reference{
+				Type: "EventHub",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.EventHub/namespaces/namespace1/eventhubs/eventhub1/authorizationRules/rule1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Eventhub",
+			"namespaces", "name",
+			"eventhubs", "name",
+			"authorizationRules", "name",
```

Comment:
Are we okay with the capital `R` here (comparing it with the discussion on the `consumergroups` URL path component)?

Reply:
It is a discrepancy with another case, it will fail with lowercase `r` here, so we face inconsistent behavior here